routes/api/authRouter.js

```diff
@@ -0,0 +1,111 @@
+const express = require("express");
+const User = require("../../models/user.js");
+const jwt = require("jsonwebtoken");
+const Joi = require("joi");
+const authMiddleware = require("../../middleware/jwt.js");
+
+const router = express.Router();
+
+const signupSchema = Joi.object({
+  email: Joi.string().email().required(),
+  password: Joi.string().required(),
+});
+
+const loginSchema = Joi.object({
+  email: Joi.string().email().required(),
+  password: Joi.string().required(),
+});
+
+router.post("/signup", async (req, res, next) => {
+  const { error } = signupSchema.validate(req.body);
+  if (error) {
+    return res.status(400).json({ message: error.details[0].message });
+  }
+
+  const { email, password } = req.body;
+  const user = await User.findOne({ email }, { _id: 1 }).lean();
+  if (user) {
+    return res.status(409).json({ message: "Email in use" });
+  }
+  try {
+    const newUser = new User({ email, password });
+    await newUser.setPassword(password);
+    await newUser.save();
+    return res.status(201).json({
+      user: {
+        email: email,
+        subscription: newUser.subscription,
+      },
+    });
+  } catch (e) {
+    next(e);
+  }
+});
+
+router.post("/login", async (req, res, next) => {
+  const { error } = loginSchema.validate(req.body);
+  if (error) {
+    return res.status(400).json({ message: error.details[0].message });
+  }
+  const { email, password } = req.body;
+
+  try {
+    const user = await User.findOne({ email });
+
+    if (!user) {
+      return res.status(401).json({ message: "No such user" });
+    }
+
+    const isPasswordCorrect = await user.validatePassword(password);
+    if (isPasswordCorrect) {
+      const payload = {
+        id: user._id,
+        email: user.email,
+        subscription: user.subscription,
+      };
+      const token = jwt.sign(payload, process.env.SECRET, { expiresIn: "12h" });
+      user.token = token;
+      await user.save();
+
+      return res.status(200).json({
+        token: token,
+        user: {
+          email: user.email,
+          subscription: user.subscription,
+        },
+      });
+    } else {
+      return res.status(401).json({ message: "Email or password is wrong" });
+    }
+  } catch (err) {
+    next(err);
+  }
+});
+
+router.get("/logout", authMiddleware, async (req, res, next) => {
+  try {
+    const userId = res.locals.user._id;
+    const user = await User.findById(userId);
```

Comment:
Moze uzyj find by id and update. BO wiadomo ze user jest wiec tylko trzeba mu token ustawic. A mongoose ma taka jedna metodą fajną.